src/utils/quickPickFile.ts

```diff
@@ -34,14 +34,36 @@ function createFileItem(rootFolder: vscode.WorkspaceFolder, uri: vscode.Uri): It
     };
 }
 
-async function resolveFileItemsInternal(rootFolder: vscode.WorkspaceFolder, fileUri: vscode.Uri | undefined, globPatterns: string[], message: string, canPickMany: boolean)
-    : Promise<Item[] | undefined> {
-    if (fileUri) {
-        return [createFileItem(rootFolder, fileUri)];
+function getDockerFileGlobPatterns(): string[] {
+    return getGlobPatterns([DOCKERFILE_GLOB_PATTERN], /^dockerfile$/i);
+}
+
+function getDockerComposeFileGlobPatterns(): string[] {
+    return getGlobPatterns([COMPOSE_FILE_GLOB_PATTERN], /^yaml$/i);
+}
+
+function getGlobPatterns(globPatterns: string[], fileTypeRegEx: RegExp): string[] {
+    const result: string[] = globPatterns;
+    try {
+        const config = vscode.workspace.getConfiguration('files').get<{}>('associations');
+        if (config) {
+            for (const globPattern of Object.keys(config)) {
+                const fileType = <string | undefined>config[globPattern];
+                if (fileType && fileTypeRegEx.test(fileType)) {
+                    result.push(globPattern);
+                }
+            }
+        }
+    } catch {
+        // ignore and use default
     }
+    return result;
+}
 
+async function resolveFilePatterns(rootFolder: vscode.WorkspaceFolder, filePatterns: string[])
```

Comment:
```suggestion
async function resolveFilesOfPattern(rootFolder: vscode.WorkspaceFolder, filePatterns: string[])
```
Since we're getting files matching the given pattern(s), I think this is a better name. I realize this was the old name, but this file has always been super confusing and bad names didn't help...